BSB_lan.ino

```diff
@@ -7787,7 +7794,7 @@ uint8_t pps_offset = 0;
           }
         }
       }
-      MQTTClient->disconnect();
+      //MQTTClient->disconnect();   //Luposoft: no needing to disconnect anymore
```

Comment:
could this be the right way?
#ifndef MQTT
  if(MQTTClient <> NULL)  
  {if (MQTTClient->connected()) {MQTTClient->disconnect()};}
#endif

Reply:
#ifndef MQTT not needed beacuse MQTTClient->disconnect()} placed inside 
```
#ifdef MQTT
...
#endif
```
block but yes
if(MQTTClient != NULL && !mqtt_mode)
{if (MQTTClient->connected()) {MQTTClient->disconnect()};}

Main problem in Ehernet Shielid is allowing 4 sockets simultaneously, now it can be:
HTTP/JSON server
MAX! devices handler
Telnet server
MQTT client/server
Update checker

Reply:
I never thought to use my guess inside #ifdef MQTT. I forgot to explain it.
Instaed using condition #ifndef MQTT you think !mqtt_mode is the better way?
Ok, I'll do it so.

Reply:
In version 2.x definitions allow/decline to compile module. mqtt_mode  can enable/disable compiled module.
If module MQTT not compiled then mqtt_mode has no sense. If MQTT compiled but mqtt_mode == 0 then we don't need disconnect because we are not connected. (may we need to add condition to connect function)